api_tests/src/actors/actor.ts

```diff
@@ -524,6 +534,40 @@ export class Actor {
         });
     }
 
+    private async assertLedgerEventPresence(
+        ledgerRel: "alpha" | "beta",
+        step: Step
+    ): Promise<void> {
+        await this.pollCndUntil(this.swap.self, (swapResponse) => {
```

Comment:
We don't really need to `pollUntil` here because we are always call this after `assertLedgerStatus` which I think already uses `pollUntil`.

That would allow you to write this logic in a simpler manner and actually have an assertion in here :)
Strictly speaking, this doesn't assert anything.

It is tricky to balance the trade-offs here.

On the one hand, we don't want to have fixed sleeps because it makes the test flaky but just going once to `cnd` and asserting on the response would be nicer in the test.

One of the reasons for this is that our relationship with cnd is one-way: We only read from `cnd` but we never "write" to it and trigger state changes ourselves. Doing that would give us a fixed point after we can assert.

Probably not worth changing something about this now, but maybe you or someone else finds these thoughts interesting :)